solr/core/src/java/org/apache/solr/response/BinaryResponseWriter.java

```diff
@@ -59,12 +60,6 @@ public void write(OutputStream out, SolrQueryRequest req, SolrQueryResponse resp
     }
   }
 
-  @Override
-  public void write(Writer writer, SolrQueryRequest request, SolrQueryResponse response)
-      throws IOException {
-    throw new RuntimeException("This is a binary writer , Cannot write to a characterstream");
```

Comment:
The existence of this was a code-smell that the hierarchy was wrong; promoting my efforts here